utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -128,31 +237,60 @@ public isolated function mapCcdaTelecomToFhirTelecom(xml telecomElement) returns
 
 # Map C-CDA name to FHIR HumanName.
 #
-# + nameElement - C-CDA name element
+# + nameElements - C-CDA name element
 # + return - Return FHIR HumanName
-public isolated function mapCcdaNametoFhirName(xml nameElement) returns r4:HumanName? {
-    xml familyElement = nameElement/<v3:family|family>;
-    xml givenElement = nameElement/<v3:given|given>;
+public isolated function mapCcdaNametoFhirName(xml nameElements) returns r4:HumanName[]? {
+    r4:HumanName[] humanNames = [];
+
+    foreach xml nameElement in nameElements {    
+        xml familyElement = nameElement/<v3:family|family>;
+        xml givenElements = nameElement/<v3:given|given>;
+        xml prefixlements = nameElement/<v3:prefix|prefix>;
+        xml suffixElements = nameElement/<v3:suffix|suffix>;
+
+
+        string family = familyElement.data().trim();
+
+        string[] given = [];
+        foreach xml givenElement in givenElements {
+            given.push(givenElement.data().trim());
+        }
+
+        string[] prefix = [];
+        foreach xml prefixlement in prefixlements {
+            prefix.push(prefixlement.data().trim());
+        }
+        
+        string[] suffix = [];
+        foreach xml suffixElement in suffixElements {
+            suffix.push(suffixElement.data().trim());
+        }
 
-    string given = givenElement.data().trim();
-    string family = familyElement.data().trim();
 
-    if given != "" || family != "" {
-        return {given: [given], family};
+        if family != "" || given.length() > 0 || prefix.length() > 0 || suffix.length() > 0 {
+            r4:HumanName name = {given: given, family, prefix, suffix};
+            humanNames.push(name);
```

Comment:
This check will give empty arrays and strings to the datatype fields. Can you try something like below?

```suggestion
        r4:HumanName name = {given: given.length()>0:given?(), family != ""?family:(), prefix: prefix.length()>0:prefix?(), suffix.length > 0:suffix?()};
            if name != {} {
                  humanNames.push(name);
            
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/e3ab4e065f72ff3ac877739a5159f37231d8bb14